Removing utterance from corpus, topology changes for punctuation, unk language model

aarora8 opened the pull request:
Adding Jonathan’s changes for hmm topo for punctuations and running on kaldi 5.4 (for new tdnn changes). Adding unk language model. only removing utterance (dev and eval) from lob corpus.  adding l2-regularize, Yiwen's slant correction , using most frequent 50 thousand words for lexicon and 190k words for unk model.

aarora8 committed:
adding aachen split and removing sentences from corpus

aarora8 committed:
minor change

aarora8 committed:
adding changes for diff hmm topo

aarora8 committed:
minor change

aarora8 committed:
minor change

aarora8 committed:
Merge remote-tracking branch 'upstream/master' into iam_new

aarora8 committed:
syncing code with local code

aarora8 committed:
Merge branch 'master' of https://github.com/kaldi-asr/kaldi into iam_new

aarora8 committed:
modification for removing irrelevent files

aarora8 committed:
minor change

aarora8 committed:
minor change

hhadian commented on egs/iam/v1/run.sh:
```diff
@@ -44,7 +44,7 @@ fi
   echo "$0: Preparing dictionary and lang..."
   local/prepare_dict.sh
-  utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8 --sil-prob 0.95 \
+  local/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8 --sil-prob 0.95 \
```
What have you changed in prepare_lang.sh?

aarora8 replied:
it is calling local ->gen_topo file, to have different number of hmm states for punctuations.

hhadian replied:
Do we actually need to do it for GMM systems too?
I imagine it does not matter much what topology we use for GMM models and we can use this special topology only for chain training.

aarora8 replied:
Jonathan, did experiments with changing topology for GMM systems and/or chain training. WER improved  if changing topology for GMM systems was changed. I took the modifications from Jonathan and added it, as it was improving the WER.  Maybe he can share more regarding the experiments.

ChunChiehChang replied:
Different topologies for the chain training did not improve the WER. Only the changing the GMM topology improves the WER. From previous experiments the better GMM topology improved the chain training results from WER 5.88% to 5.32%. These number are from past results with LOB corpus included. I haven't run experiments to compare the results with the Wellington corpus included.

hhadian commented:
I guess this change in GMM-based models won't really affect the chain
training. How much does the new topology improve gmm-based WERs and CERs?

On Feb 23, 2018 2:49 AM, "Ashish Arora" <notifications@github.com> wrote:

*@aarora8* commented on this pull request.
------------------------------

In egs/iam/v1/run.sh
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>:

> @@ -44,7 +44,7 @@ fi
 if [ $stage -le 2 ]; then
   echo "$0: Preparing dictionary and lang..."
   local/prepare_dict.sh
-  utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \
+  local/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \

Jonathan, did experiments with changing topology for GMM systems and/or
chain training. WER improved if changing topology for GMM systems was
changed. I took the modifications from Jonathan and added it, as it was
improving the WER. Maybe he can share more regarding the experiments.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AOW_DeI4JpIvWAjzDImRkledJQqInXkXks5tXfYHgaJpZM4SPyKM>
.

aarora8 commented:
Best results with different GMM topology is WER/CER:

%WER 12.39 [ 2297 / 18542, 217 ins, 460 del, 1620 sub ]

%WER 6.01 [ 3965 / 65921, 740 ins, 925 del, 2300 sub ]



Without different GMM topology and new tdnn changes, best result is

%WER 13.96 [ 2589 / 18542, 251 ins, 671 del, 1667 sub ]

%WER 7.49 [ 4940 / 65921, 773 ins, 1848 del, 2319 sub ]


I will run a new code with new tdnn changes but without different topology.

Thanks,

Ashish


________________________________
From: Hossein Hadian <notifications@github.com>
Sent: Thursday, February 22, 2018 7:23:44 PM
To: hhadian/kaldi
Cc: Ashish Arora; Mention
Subject: Re: [hhadian/kaldi] Removing utterance from corpus, topology changes for punctuation, unk language model (#13)

I guess this change in GMM-based models won't really affect the chain
training. How much does the new topology improve gmm-based WERs and CERs?

On Feb 23, 2018 2:49 AM, "Ashish Arora" <notifications@github.com> wrote:

*@aarora8* commented on this pull request.
------------------------------

In egs/iam/v1/run.sh
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>:

> @@ -44,7 +44,7 @@ fi
if [ $stage -le 2 ]; then
echo "$0: Preparing dictionary and lang..."
local/prepare_dict.sh
- utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \
+ local/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \

Jonathan, did experiments with changing topology for GMM systems and/or
chain training. WER improved if changing topology for GMM systems was
changed. I took the modifications from Jonathan and added it, as it was
improving the WER. Maybe he can share more regarding the experiments.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AOW_DeI4JpIvWAjzDImRkledJQqInXkXks5tXfYHgaJpZM4SPyKM>
.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/hhadian/kaldi/pull/13#issuecomment-367868700>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AcFBRZXGCAo19aJXs9Qn9N-kEgkImJ1eks5tXgUQgaJpZM4SPyKM>.

aarora8 commented:
sorry, I made a mistake in last mail.
Best results with different GMM topology and new tdnn changes is WER/CER:
%WER 12.39 [ 2297 / 18542, 217 ins, 460 del, 1620 sub ]

%WER 6.01 [ 3965 / 65921, 740 ins, 925 del, 2300 sub ]



Without different GMM topology but old tdnn changes, best result is
%WER 13.96 [ 2589 / 18542, 251 ins, 671 del, 1667 sub ]

%WER 7.49 [ 4940 / 65921, 773 ins, 1848 del, 2319 sub ]

I will run a new code with new tdnn changes but without different topology.
Thanks,
Ashish


________________________________
From: Hossein Hadian <notifications@github.com>
Sent: Thursday, February 22, 2018 7:23:44 PM
To: hhadian/kaldi
Cc: Ashish Arora; Mention
Subject: Re: [hhadian/kaldi] Removing utterance from corpus, topology changes for punctuation, unk language model (#13)

I guess this change in GMM-based models won't really affect the chain
training. How much does the new topology improve gmm-based WERs and CERs?

On Feb 23, 2018 2:49 AM, "Ashish Arora" <notifications@github.com> wrote:

*@aarora8* commented on this pull request.
------------------------------

In egs/iam/v1/run.sh
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>:

> @@ -44,7 +44,7 @@ fi
if [ $stage -le 2 ]; then
echo "$0: Preparing dictionary and lang..."
local/prepare_dict.sh
- utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \
+ local/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8
--sil-prob 0.95 \

Jonathan, did experiments with changing topology for GMM systems and/or
chain training. WER improved if changing topology for GMM systems was
changed. I took the modifications from Jonathan and added it, as it was
improving the WER. Maybe he can share more regarding the experiments.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/hhadian/kaldi/pull/13#discussion_r170124401>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AOW_DeI4JpIvWAjzDImRkledJQqInXkXks5tXfYHgaJpZM4SPyKM>
.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/hhadian/kaldi/pull/13#issuecomment-367868700>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AcFBRZXGCAo19aJXs9Qn9N-kEgkImJ1eks5tXgUQgaJpZM4SPyKM>.

aarora8 commented:
Best results with different GMM topology 
%WER 12.39 [ 2297 / 18542, 217 ins, 460 del, 1620 sub ]
%WER 6.01 [ 3965 / 65921, 740 ins, 925 del, 2300 sub ]

Without different GMM topology 
%WER 12.76 [ 2366 / 18542, 226 ins, 472 del, 1668 sub ]
%WER 6.14 [ 4049 / 65921, 743 ins, 941 del, 2365 sub ]

Thanks,

aarora8 commented:
**results with different GMM topology**
%WER 12.39 [ 2297 / 18542, 217 ins, 460 del, 1620 sub ]
%WER 6.01 [ 3965 / 65921, 740 ins, 925 del, 2300 sub ]

%WER 12.76 [ 2366 / 18542, 226 ins, 472 del, 1668 sub ]
%WER 6.14 [ 4049 / 65921, 743 ins, 941 del, 2365 sub ]

**Without different GMM topology**
%WER 13.66 [ 2533 / 18542, 201 ins, 667 del, 1665 sub ]
%WER 6.89 [ 4544 / 65921, 683 ins, 1600 del, 2261 sub ]

%WER 13.24 [ 2455 / 18542, 201 ins, 573 del, 1681 sub ]
%WER 6.63 [ 4368 / 65921, 699 ins, 1340 del, 2329 sub ]

aarora8 committed:
adding changes for pull request
